lib/puppet/resource/catalog.rb

```diff
@@ -86,13 +108,12 @@ def container_of(resource)
     adjacent(resource, :direction => :in)[0]
   end
 
-  def add_one_resource(resource)
-    title_key = title_key_for_ref(resource.ref)
+  def add_one_resource(resource, title_key, idx)
```

Comment:
`add_resource_before` and `add_resource_after`, but it looks like they don't actually need the title key. This might be cruft from an older version of the code, TBH. I'll clean it up